Rakefile

```diff
@@ -8,3 +8,56 @@ end
 
 desc "Run tests"
 task :default => :test
+
+def program_exists?(program)
+  system '/usr/bin/which', program, [:out,:err]=>'/dev/null'
+end
+
+desc "Check for Node.js and NPM packages"
+task :check_for_node do
+  which = '/usr/bin/which'
+  opts = {[:out,:err]=>'/dev/null'}
+
+  unless system(which, 'which', opts)
+    puts [
+      "Cannot automatically check for Node.js and NPM packages on this system.",
+      "If Node.js is not installed, visit https://nodejs.org/.",
+      "If any of the following packages are not yet installed, please install",
+      "them by executing `npm install -g PACKAGE`, where `PACKAGE` is one of",
+      "the names below:"].join("\n")
```

Comment:
FYI, heredocs may be of interest.

http://en.wikibooks.org/wiki/Ruby_Programming/Here_documents